### PR TITLE
Remove GH cache after PR is merged

### LINKED ISSUE #938: Remove cache when a branch gets deleted
### Context
Github will start to evict caches once [the total size is over 10G](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy).

We have these 2 caches:
```
cargo-<branch-name>-<hash>
sccache-<branch-name>-<hash>
```
See https://github.com/litentry/litentry-parachain/actions/caches

Caches will be inevitably generated during PR creation and merging. When a PR gets closed and the branch gets deleted, we'd like to automatically delete the corresponding caches to make as much room as possible

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by BillyWooo**
This action workflow will remove the PR branch related caches after the PR is closed.

resolves #938

### CODE REVIEW

File: .github/workflows/rm_gh_cache_after_pr_closed.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Cleanup caches after PR closed
```
Comment (Kailai-Wang) [inline .github/workflows/rm_gh_cache_after_pr_closed.yml]: Let's rename the file with hyphens: `x-y-z.yml` (to be consistent with other files)

File: .github/workflows/rm_gh_cache_after_pr_closed.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+on:
+  pull_request:
+    # branches:
+    #   - dev
```
Comment (Kailai-Wang) [inline .github/workflows/rm_gh_cache_after_pr_closed.yml]: So this action can be triggered on PRs towards any branch, is that intended? If so let's remove the commented code

File: .github/workflows/rm_gh_cache_after_pr_closed.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    types:
+      - closed
+  workflow_dispatch:
```
Comment (Kailai-Wang) [inline .github/workflows/rm_gh_cache_after_pr_closed.yml]: What's the branch if it's triggered by `workflow_dispatch` (I assume it's null), then run logic will be broken.

File: .github/workflows/rm_gh_cache_after_pr_closed.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    runs-on: ubuntu-latest
+    permissions: write-all
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Extract branch name
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+        id: extract_branch
```
Comment (Kailai-Wang) [inline .github/workflows/rm_gh_cache_after_pr_closed.yml]: Why do we need this step?

File: .github/workflows/rm_gh_cache_after_pr_closed.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+          BRANCH=${{ github.head_ref }}
+
+          cacheKeysForPR=$(gh actions-cache list -R $REPO | cut -f 1 )
+
+          ## Setting this to not fail the workflow while deleting cache keys.
+          set +e
+          echo "Deleting caches after PR closed for branch=$BRANCH ..."
+          for cacheKey in $cacheKeysForPR
+          do
+              if [[ $cacheKey == *"$BRANCH"* ]]
```
Comment (Kailai-Wang) [inline .github/workflows/rm_gh_cache_after_pr_closed.yml]: Can we simplify it by `grep` in line 30?

**Review by Kailai-Wang — CHANGES_REQUESTED** [review]
Have some questions. Btw is this GHA tested somewhere? If so we can put the link here

**Comment by BillyWooo** [comment]
> Have some questions. Btw is this GHA tested somewhere? If so we can put the link here

Yes. Tested here: https://github.com/BillyWooo/litentry-parachain/actions/runs/3942679464/jobs/6746609073
Also here: https://github.com/BillyWooo/litentry-parachain/actions/runs/3947330343/jobs/6756029090

File: .github/workflows/rm_gh_cache_after_pr_closed.yml (hunk 1 above)
Comment (BillyWooo) [reply .github/workflows/rm_gh_cache_after_pr_closed.yml]: Oh, yes.

File: .github/workflows/rm_gh_cache_after_pr_closed.yml (hunk 2 above)
Comment (BillyWooo) [reply .github/workflows/rm_gh_cache_after_pr_closed.yml]: Yes. Intended. Will remove.

File: .github/workflows/rm_gh_cache_after_pr_closed.yml (hunk 4 above)
Comment (BillyWooo) [reply .github/workflows/rm_gh_cache_after_pr_closed.yml]: forgot to remove. was used for testing.

File: .github/workflows/rm_gh_cache_after_pr_closed.yml (hunk 5 above)
Comment (BillyWooo) [reply .github/workflows/rm_gh_cache_after_pr_closed.yml]: Yes.

File: .github/workflows/rm_gh_cache_after_pr_closed.yml (hunk 3 above)
Comment (BillyWooo) [reply .github/workflows/rm_gh_cache_after_pr_closed.yml]: Yes. Should be removed. Not necessary.

**Review by Kailai-Wang — APPROVED** [review]
Thank you, we can test it in real scenarios

**Comment by Kailai-Wang** [comment]
I think we still need a small adjustment for this one:
```
cacheKeysForPR=$(gh actions-cache list -R $REPO | cut -f 1 | grep "$BRANCH")
```
it's possible that `grep` finds nothing because the target branch didn't produce any caches, like this run:
https://github.com/litentry/litentry-parachain/actions/runs/3948688455

but the exit code is 1

I'll submit a minor fix for this

**Pull request merged**